### DISCUSSION TITLE
It seems that you cannot access public methods inside a UserControl from the outside, which makes it impossible to perform unit testing. Am I understanding this correctly?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15817] [Q&A]
Or is there a better way to perform unit testing on methods within a UserControl?

```
 // Start Headless session passing Application type.
 using var session = HeadlessUnitTestSession.StartNew(typeof(App));

  _ = await session.Dispatch(async Task<bool> () =>
 {

     UserControl step2UC = new Step2UcClass();
          
     step2UC.TestMethod(); //Can't find TestMethod

     return true;
 }, CancellationToken.None);
```

**Comment by maxkatz6** [comment]
You should be able to access public methods just fine. It works by .NET members visibility rules, as per usual.

**Reply by IronEaterBeast** [reply]
Okay, I understand my mistake now. The public method I wanted to access is not in the parent UserControl.
`
UserControl step2UC = new Step2UcClass();
`

**Reply by timunie** [reply]
sure UserControl is the base class of StepUcClass. so UserControl don't know about properties of it's inheritors....
